packages/ui/src/Wiz.svelte

```diff
@@ -0,0 +1,246 @@
+<script lang="ts">
+  import UserAvatar from './icons/UserAvatar.svelte';
+  import WizAvatar from './icons/WizAvatar.svelte'
+  import WizIcon from './icons/WizIcon.svelte'
+  import XIcon from './icons/XIcon.svelte';
+  import ExperimentalTooltip from './ExperimentalTooltip.svelte';
+  import type { GenericOptions } from '@openzeppelin/wizard';
+  import { nanoid } from 'nanoid';
+  import MinimizeIcon from './icons/MinimizeIcon.svelte';
+  import MaximizeIcon from './icons/MaximizeIcon.svelte';
+
+  export let functionCall: {
+    name?: string,
+    opts?: any
+  }
+  export let currentOpts: Required<GenericOptions>
+
+  interface Chat {
+    role: 'user' | 'assistant' | 'system'
+    content: string
+  }
+
+  const chatId = nanoid()
+  let inProgress = false
+  let currentMessage = ''
+  let showing: boolean = true
+  let expanded: boolean = false
+  let messages: Chat[] = [{
+    role: 'assistant',
+    content: 'I can also edit Wizard settings directly, so from time to time I will update those based on your input.',
+  }, {
+    role: 'assistant',
+    content: 'Wiz here 👋. Feel free to ask any questions you have about smart contract development.',
+  }]
+
+  const nameMap = {
+    erc20: 'ERC20',
+    erc721: 'ERC721',
+    erc1155: 'ERC1155',
+    governor: 'Governor',
+    custom: 'Custom',
+  }
+
+  const sampleMessages = [
+    'Make a token with supply of 10 million',
+    'What does mintable do?',
+    'Make a contract for a DAO',
+  ]
+
+  const addMessage = (message: Chat) => {
+    messages = [{
+      role: message.role,
+      content: message.content
+    }, ...messages]
+  }
+
+  const submitChat = (message: Chat) => {
+    inProgress = true
+    addMessage(message)
+    input = ''
+
+    const chat = messages.slice(0, messages.length - 2).reverse()
+
+    fetch('/ai', {
+      method: "POST",
+      headers: {
+        "Content-Type": "application/json",
+      },
+      body: JSON.stringify({
+        currentOpts,
+        chatId,
+        messages: chat,
+        stream: true,
+      }),
+    }).then(async (response) => {
+      const reader = response.body?.getReader();
+      const decoder = new TextDecoder("utf-8");
+      let result = '';
+      while (true && reader) {
+        const { done, value } = await reader.read();
+        if (done) {
+          break;
+        }
+        // Massage and parse the chunk of data
+        const chunk = decoder.decode(value);
+        result += chunk
+
+        if (result.startsWith('{"function_call":')) {
+          currentMessage = 'Executing function...'
+        }
+        else {
+          currentMessage = result
+        }
+      }
```

Comment:
Since reader is not being changed, it would be clearer to have an if-statement and then a nested while-loop:
```suggestion
      if (reader) {
        while (true) {
          const { done, value } = await reader.read();
          if (done) {
            break;
          }
          // Massage and parse the chunk of data
          const chunk = decoder.decode(value);
          result += chunk
  
          if (result.startsWith('{"function_call":')) {
            currentMessage = 'Executing function...'
          }
          else {
            currentMessage = result
          }
        }
      }
```

Reply:
Applied